test/src/test/java/org/corfudb/runtime/object/transactions/OptimisticTXConcurrencyTest.java

```diff
@@ -0,0 +1,185 @@
+package org.corfudb.runtime.object.transactions;
+
+import lombok.Getter;
+import org.corfudb.runtime.exceptions.TransactionAbortedException;
+import org.corfudb.runtime.object.CorfuSharedCounter;
+import org.corfudb.runtime.view.AbstractViewTest;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Created by dalia on 12/8/16.
+ */
+public class OptimisticTXConcurrencyTest extends AbstractViewTest {
+    @Getter
+    final String defaultConfigurationString = getDefaultEndpoint();
+
+    @Before
+    public void before() { getDefaultRuntime(); }
+
+    @Test
+    public void testOpacity() throws Exception {
+
+        int n = 5;
+        assertThat(n).isGreaterThan(1); // don't change concurrency to less than 2, test will break
+
+        CorfuSharedCounter[] sharedCounters = new CorfuSharedCounter[n];
+
+        for (int i = 0; i < n; i++)
+          sharedCounters[i] = getRuntime().getObjectsView()
+                  .build()
+                  .setStreamName("test"+i)
+                  .setType(CorfuSharedCounter.class)
+                  .open();
+
+        // initialize all shared counters
+        for (int i = 0; i < n; i++)
+            sharedCounters[i].setValue(32);
```

Comment:
When #374 gets merged, this won't work anymore. Can you move these values into a constant - `static final int SOME_USEFUL_NAME = 32`?